server/game/core/Player.js

```diff
@@ -288,16 +288,33 @@ class Player extends GameObject {
     }
 
     /**
-     * Returns if a card is in play (unots, upgrades, provinces, holdings) that has the passed trait
+     * Returns if a card is in play (units, upgrades, provinces, holdings) that has the passed trait
      * @param {string} trait
-     * @returns {boolean} true/false if the trait is in pay
+     * @param {any} ignoreUnit
+     * @returns {boolean} true/false if the trait is in play
      */
-    isTraitInPlay(trait) {
+    isTraitInPlay(trait, ignoreUnit = undefined) {
+        if (ignoreUnit !== undefined) {
```

Comment:
Could do this all in one statement using a ternary operator, i.e.

```javascript
return ignoreUnit != undefined
    ? this.getOtherUnitsInPlay(ignoreUnit).some((card) => card.hasSomeTrait(trait))
    : this.getUnitsInPlay().some((card) => card.hasSomeTrait(trait));
```

Also, this is the one situation it's better to use `!=` instead of `!==` because `!=` will match both `null` and `undefined`

Ditto on both of the above for `isAspectInPlay`